sqlite/src/func.c

```diff
@@ -796,11 +798,16 @@ static void comdb2CtxinfoFunc(
   if( sqlite3_value_type(argv[0])!=SQLITE_TEXT ){
     return;
   }
+
+  struct sql_thread *thd = pthread_getspecific(query_info_key);
+  struct sqlclntstate *clnt = thd!=NULL ? thd->clnt : NULL;
+
   zName = (const char *)sqlite3_value_text(argv[0]);
   if( sqlite3_stricmp(zName, "parallel")==0 ){
-    struct sql_thread *thd = pthread_getspecific(query_info_key);
-    struct sqlclntstate *clnt = thd!=NULL ? thd->clnt : NULL;
     sqlite3_result_int(context, clnt!=NULL && clnt->conns!=NULL);
+  } else if( sqlite3_stricmp(zName, "user")==0 ){
+    sqlite3_result_text(context, (clnt!=NULL) ? get_current_user(clnt) : "",
```

Comment:
If the clnt is NULL or the return value of get_current_user is NULL, simply skipping the call to sqlite3_result_text may be good here as that would return a SQL NULL to the caller?